metrics/core/metrics.go

```diff
@@ -741,6 +755,111 @@ var MetricFilesystemInodesFree = Metric{
 	},
 }
 
+var MetricAcceleratorMemoryCapacity = Metric{
+	MetricDescriptor: MetricDescriptor{
+		Name:        "accelerator/memory_capacity",
+		Description: "Memory capacity of an accelerator",
+		Type:        MetricGauge,
+		ValueType:   ValueInt64,
+		Units:       UnitsBytes,
+		Labels:      metricLabels,
+	},
+	HasLabeledMetric: func(spec *cadvisor.ContainerSpec, stat *cadvisor.ContainerStats) bool {
+		if len(stat.Accelerators) == 0 {
+			return false
+		}
+
+		return true
+	},
+	GetLabeledMetric: func(spec *cadvisor.ContainerSpec, stat *cadvisor.ContainerStats) []LabeledMetric {
+		result := make([]LabeledMetric, 0, len(stat.Accelerators))
+		for _, ac := range stat.Accelerators {
+			result = append(result, LabeledMetric{
+				Name: "accelerator/memory_capacity",
+				Labels: map[string]string{
+					LabelResourceID.Key: fmt.Sprintf("%s:%s:%s", ac.Make, ac.Model, ac.ID),
+				},
+				MetricValue: MetricValue{
+					ValueType:  ValueInt64,
+					MetricType: MetricGauge,
+					IntValue:   int64(ac.MemoryTotal),
+				},
+			})
+		}
+		return result
+	},
+}
+
+var MetricAcceleratorMemoryUsed = Metric{
+	MetricDescriptor: MetricDescriptor{
+		Name:        "accelerator/memory_usage",
+		Description: "Memory used of an accelerator",
+		Type:        MetricGauge,
+		ValueType:   ValueInt64,
+		Units:       UnitsBytes,
+		Labels:      metricLabels,
+	},
+	HasLabeledMetric: func(spec *cadvisor.ContainerSpec, stat *cadvisor.ContainerStats) bool {
+		if len(stat.Accelerators) == 0 {
+			return false
+		}
+
+		return true
+	},
+	GetLabeledMetric: func(spec *cadvisor.ContainerSpec, stat *cadvisor.ContainerStats) []LabeledMetric {
+		result := make([]LabeledMetric, 0, len(stat.Accelerators))
+		for _, ac := range stat.Accelerators {
+			result = append(result, LabeledMetric{
+				Name: "accelerator/memory_usage",
+				Labels: map[string]string{
+					LabelResourceID.Key: fmt.Sprintf("%s:%s:%s", ac.Make, ac.Model, ac.ID),
+				},
+				MetricValue: MetricValue{
+					ValueType:  ValueInt64,
+					MetricType: MetricGauge,
+					IntValue:   int64(ac.MemoryUsed),
+				},
+			})
+		}
+		return result
+	},
+}
+
+var MetricAcceleratorUtility = Metric{
+	MetricDescriptor: MetricDescriptor{
+		Name:        "accelerator/utility",
+		Description: "Utility of an accelerator",
+		Type:        MetricGauge,
+		ValueType:   ValueInt64,
+		Units:       UnitsCount,
+		Labels:      metricLabels,
+	},
+	HasLabeledMetric: func(spec *cadvisor.ContainerSpec, stat *cadvisor.ContainerStats) bool {
+		if len(stat.Accelerators) == 0 {
+			return false
+		}
+
+		return true
+	},
+	GetLabeledMetric: func(spec *cadvisor.ContainerSpec, stat *cadvisor.ContainerStats) []LabeledMetric {
+		result := make([]LabeledMetric, 0, len(stat.Accelerators))
+		for _, ac := range stat.Accelerators {
+			result = append(result, LabeledMetric{
+				Name: "accelerator/utility",
+				Labels: map[string]string{
+					LabelResourceID.Key: fmt.Sprintf("%s:%s:%s", ac.Make, ac.Model, ac.ID),
```

Comment:
Am I reading this right? This is just one label with key=LabelResourceID.Key and value=concatenation of Make, Model and ID? If so, I don't think we want this.

All these metrics should have three extra labels.
make, model and accelerator_id.

Reply:
Will do.